src/lib/Transport/index.ts

```diff
@@ -80,66 +80,81 @@ export default class Transport {
     });
   }
 
-  
-  build(req: Request): Promise<request.CoreOptions> {
-    if (!this.isValidRequest(req.url)) {
-      const err = new Error('url provided is not a valid domo app endpoint');
-      
-      return Promise.reject(err);
-    }
-    
+  createContext(): Promise {
+    const options = {
+      method: 'POST',
+      url: `${this.client.server}/domoapps/apps/v2/contexts`,
+      json: { designId: this.manifest.id, mapping: this.manifest.mapping },
+      headers: this.client.getAuthHeader(),
+    };
+
+    return new Promise((resolve, reject) => {
+      request(options, (error, response, body) => {
+        if (error) reject(error);
+
+        if (response.statusCode !== 200) reject(response);
+
+        resolve(body[0]);
+      });
+    });
+  }
+
+  build(req: IncomingMessage): Promise<request.CoreOptions> {
     let api: string;
-    
+
     return this.domainPromise
       .then((domain) => {
         api = `${domain}${req.url}`;
-        
+
         return this.createContext();
       })
-      .then((context) => {
+      .then((context, body) => {
         const jar = request.jar();
-        
+
         const referer = (req.headers.referer.indexOf('?') >= 0)
           ? (`${req.headers.referer}&context=${context.id}`)
           : (`${req.headers.referer}?userId=27&customer=dev&locale=en-US&platform=desktop&context=${context.id}`);
-        
+
         const headers = {
-          ...req.headers,
           ...this.client.getAuthHeader(),
-          referer,
+          'Content-Type': req.headers['content-type'] || req.headers['Content-Type'] || 'application/json',
+          Referer: referer,
         };
-        
-        return {
+
+        const options = {
           jar,
           headers,
           url: api,
           method: req.method,
-          body: req.body,
+          body: null,
         };
-      })
-      .catch((err) => {
-        throw new DomoException(err, req.url);
+
+        return this.parseBody(req).then((body) => {
+          options.body = body;
+
+          return options;
+        });
       });
   }
-  
-  createContext(): Promise {
-    const options = {
-      method: 'POST',
-      url: `${this.client.server}/domoapps/apps/v2/contexts`,
-      json: { designId: this.manifest.id, mapping: this.manifest.mapping },
-      headers: this.client.getAuthHeader(),
-    };
-    
-    return new Promise((resolve, reject) => {
-      request(options, (error, response, body) => {
-        const ok = 200;
-        
-        if (error) reject(error);
-        
-        if (response.statusCode !== ok) reject(response);
-        
-        resolve(body[0]);
-      });
+
+  private parseBody(req: IncomingMessage): Promise<string|void> {
+    return new Promise((resolve) => {
+      const body = [];
+
+      try {
+        req.on('data', chunk => body.push(chunk));
+
+        req.on('end', () => {
+          const contentType = req.headers['Content-Type'] || req.headers['content-type'];
```

Comment:
Why are we grabbing the contentType here?

Reply:
Old code. I've removed it.